gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerIntegrateTest.java

```diff
@@ -1087,4 +1088,26 @@ public void shouldGenerateTemporaryErrorResponseStatusCode() throws Exception {
             cluster.close();
         }
     }
+
+    /**
+     * Reproducer for TINKERPOP-2765.
+     */
+    @Test
+    public void shouldHandleMultipleLambdaTranslationsInParallel() throws Exception {
+        final GraphTraversalSource g = traversal().withRemote(conf);
+
+        final CompletableFuture<Traversal<Object, Object>> firstRes = g.with("evaluationTimeout", 90000L).inject(1).sideEffect(Lambda.consumer("Thread.sleep(100)")).promise(Traversal::iterate);
+        final CompletableFuture<Traversal<Object, Object>> secondRes = g.with("evaluationTimeout", 90000L).inject(1).sideEffect(Lambda.consumer("Thread.sleep(100)")).promise(Traversal::iterate);
+        final CompletableFuture<Traversal<Object, Object>> thirdRes = g.with("evaluationTimeout", 90000L).inject(1).sideEffect(Lambda.consumer("Thread.sleep(100)")).promise(Traversal::iterate);
+
+        try {
+            firstRes.get();
+            secondRes.get();
+            thirdRes.get();
+        } catch (CompletionException ce) {
```

Comment:
Out of curiosity, why are we catching CompletionException instead of general Exception? Asking because I find that occasionally in the failing case it waits for everything to complete before sending the CompletionException, even when a compilation exception was thrown immediately.

Reply:
I didn't actually notice this when running the test myself. I agree that it could just be a general Exception but I didn't want to introduce any test failures that were caused by a different, unrelated exception being thrown.